test/cypress/support/commands.js

```diff
@@ -336,6 +365,62 @@ Cypress.Commands.add('giAcceptGroupInvite', (invitationLink, {
   }
 })
 
+Cypress.Commands.add('giAcceptUsersGroupInvite', (invitationLink, {
+  usernames,
+  existingMemberUsername,
+  displayNames,
+  actionBeforeLogout,
+  groupName,
+  bypassUI
+}) => {
+  const { groupId, inviteSecret } = getParamsFromInvitationLink(invitationLink)
+  for (const username of usernames) {
+    if (bypassUI) {
+      cy.giSignup(username, { bypassUI })
+      cy.window().its('sbp').then(async sbp => {
+        await sbp('gi.actions/group/joinWithInviteSecret', groupId, inviteSecret)
+        await sbp('controller/router').push({ path: '/pending-approval' }).catch(e => {})
+      })
+    } else {
+      cy.visit(invitationLink)
+      cy.giSignup(username, { isInvitation: true, groupName })
+    }
+
+    // NOTE: checking 'data-groupId' is for waiting until joining process would be finished
+    cy.getByDT('pendingApprovalTitle').invoke('attr', 'data-groupId').should('eq', groupId)
+    // TODO: should remove the following cy.wait()
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(1000)
+
+    cy.giLogout()
+  }
+
+  cy.giLogin(existingMemberUsername, { bypassUI })
+  // TODO: should remove the following cy.wait()
+  // eslint-disable-next-line cypress/no-unnecessary-waiting
+  cy.wait(2000)
+  cy.giLogout()
```

Comment:
Actually
- The first one (`cy.wait(1000)`) is to wait for `KEY_REQUEST` event is published
- The second one (`cy.wait(2000)`) is to wait for `KEY_SHARE` event is published after getting received the `KEY_REQUEST` event

I will think about it more to remove this `cy.wait`. If you have good solution, please kindly share with me.

cc: @taoeffect, @corrideat

Reply:
@Silver-IT In Cypress, we have 2 options:

1. Wait for a change in the UI
2. Wait for an event to be emitted using `sbp`

So you'll need to pick one of these.

If there is no change in the UI, and there is no event, you can create the event yourself.

Reply:
Yeah, I know we have those two options. But in this case, there is no changes in the UI, and no events emitted.
Do you think it's fine to make events every time we publish the `KEY_REQUEST`, and `KEY_SHARE` events?

Reply:
@Silver-IT I wouldn't post an event each time anything sends a KEY_REQUEST/KEY_SHARE, but rather in the `gi.actions` selector that you're waiting to finish, at the end of it, emit an event. You can even name it `CYPRESS_FINISHED_KEY_SHARE_EVENT` so that it's clear what it's being used for